File: C Programs/Problems on String/Accept string & display it in reverse order.c
/* Write a program which accept string from user and display it in reverse order.

Input : “MarvellouS”
Output : “SuollevraM”

*/
#include<stdio.h>

void Reverse(char str[])
{
    int iStart=0;
    int iEnd=0;
    char itemp='\0';


    while(str[iEnd]!='\0')
    {
       iEnd++;;
    }
    iEnd--;

     while(iStart<iEnd)
       {
           itemp=str[iStart];
           str[iStart]=str[iEnd];
           str[iEnd]=itemp;

        iStart++;
        iEnd--;
       }
}

int main()
{
    char arr[30];

    printf("Enter string: \n");
    scanf("%[^'\n']s",arr);

    Reverse(arr);
    printf("reverse string is %s",arr);

    return 0;
}

File: Java Programs/Problems on String/Accept string & display it in reverse order.java
/* Write a java program which accept string from user and display it in reverse order.

Input : MarvellouS
Output : SuollevraM 

*/

import java.lang.*;
import java.util.*;

class DemoStringX
 {
     void Reverse(String str)
     {
         char arr[]=str.toCharArray();
         int size=arr.length;
	       int iStart=0;
         int iEnd=size-1;
         int i=0;
         char temp='\0';
            
        while(iStart<iEnd)
        {
           temp=arr[iStart];
           arr[iStart]=arr[iEnd];
           arr[iEnd]=temp;           
           iStart++;
	         iEnd--;
        }

        System.out.println(arr);
     }
 }

class Demo
   {
       public static void main(String arg[])
         {
              Scanner sobj=new Scanner(System.in);
                 
              DemoStringX dobj=new DemoStringX();
                 
              System.out.println("Enter String: ");
              String str=sobj.nextLine();

	            dobj.Reverse(str);
                 
         }
    }

